src/psyclone/psyir/transformations/profile_trans.py

```diff
@@ -211,21 +211,8 @@ def apply(self, nodes, options=None):
 
         # Create the ProfileNode. All of the supplied child nodes will have
         # the Profile's Schedule as their parent.
-        from psyclone.profiler import ProfileNode
-        profile_node = ProfileNode(parent=node_parent, children=node_list[:],
-                                   name=name)
-
-        # Correct the parent's list of children. Use a slice of the list of
-        # nodes so that we're looping over a local copy of the list. Otherwise
-        # things get confused when we remove children from the list.
-        for child in node_list[:]:
-            # Remove child from the parent's list of children
-            node_parent.children.remove(child)
-
-        # Add the Profile node as a child of the parent
-        # of the nodes being enclosed and at the original location
-        # of the first of these nodes
-        node_parent.addchild(profile_node,
-                             index=node_position)
+
+        from psyclone.psyir.nodes import ProfileNode
+        ProfileNode(parent=node_parent, children=node_list[:], name=name)
```

Comment:
Please add a comment making clear what this constructor does as I think it looks a bit confusing. (If you can think of a better way of doing it that would also be worth considering.)

Reply:
I actually like it, assuming that it is (now) documented. But am happy to redo this as part of #645.